apps/judicial-system/backend/src/app/modules/defendant/defendant.service.ts

```diff
@@ -200,7 +201,16 @@ export class DefendantService {
     caseId: string,
     defendantNationalId: string,
     update: UpdateDefendantDto,
+    user?: User,
   ): Promise<Defendant> {
+    const isDefenderChoiceConfirmed = Boolean(
+      user &&
+        isDistrictCourtUser(user) &&
+        update.isDefenderChoiceConfirmed === true,
+    )
+
+    update = { ...update, isDefenderChoiceConfirmed: isDefenderChoiceConfirmed }
+
```

Comment:
We want to make sure that no one but a court role can confirm a defender choice. If anyone else makes a change we want to retract an existing confirmation.

Reply:
Yeah you're right, I know it's not currently being done I was more so thinking of this as future proofing in case someone starts calling this in the future without knowing it shouldn't be changed by anyone but a court user, but it might be unnecessary worry on my part 😅

Reply:
Ive changed this